sw-precache.js

```diff
@@ -119,11 +120,13 @@ module.exports = function(params) {
   params.logger(util.format('Total precache size is about %s for %d resources.',
     formatBytesAsString(cumulativeSize), relativeUrls.length));
 
-  var templateBuffer = fs.readFileSync(params.templateFilePath);
-  return _.template(templateBuffer, {
+  var template = fs.readFileSync(params.templateFilePath, {encoding: 'utf8'});
+  var populatedTemplate = _.template(template, {
     handleFetch: params.handleFetch,
     importScripts: params.importScripts ? params.importScripts.map(JSON.stringify).join(',') : null,
     includeCachePolyfill: params.includeCachePolyfill,
     precacheConfig: JSON.stringify(precacheConfig)
   });
+
+  return resumer().queue(populatedTemplate).end();
```

Comment:
I think the main interface should be callback, but of course you might want to add a streaming endpoint for convenience, or even better a separate wrapper module. Though users should be able to easily use it with `gulp-file`, so might be enough to just document that.